File: python/20_Valid_Parentheses/solution.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        for ch in s:
            if ch in '([{':
                stack.append(chr(ord(ch) + 1) if ch == '(' else chr(ord(ch) + 2))
            elif not stack or ch != stack.pop():
                return False
        return not stack

File: rust/20_Valid_Parentheses/solution.rs
use std::char;

impl Solution {
    pub fn is_valid(s: String) -> bool {
        let mut stack: Vec<char> = vec![];
        for ch in s.chars() {
            match ch {
                '{' => stack.push('}'),
                '[' => stack.push(']'),
                '(' => stack.push(')'),
                ch if stack.len() == 0 || stack.pop().unwrap() != ch => return false,
                _ => (),
            }
        }
        stack.len() == 0
    }
}
